app/assets/v2/js/grants/index.js

```diff
@@ -314,6 +314,49 @@ if (document.getElementById('grants-showcase')) {
         }
         return vm.grants;
       },
+      addAllToCart: async function() {
+        let vm = this;
+
+        if (this.cart_lock)
+          return;
+
+        this.lock = true;
+
+        const base_params = {
+          no_pagination: true,
+          sort_option: this.sort,
+          network: this.network,
+          keyword: this.keyword,
+          state: this.state,
+          category: this.category,
+          type: this.current_type
+        };
+
+        if (this.following) {
+          base_params['following'] = this.following;
+        }
+
+        if (this.idle_grants) {
+          base_params['idle'] = this.idle_grants;
+        }
+
+        if (this.show_contributions) {
```

Comment:
is there a way to DRY this so that if we add a new filter, it doesnt break the add all to cart functionality?

Reply:
I think in some other opportunity we should dry the filters in general then we can reuse it here